### PR TITLE
Replace metering decorators with proxy

### PR DESCRIPTION
**Pull request by rouzwawi**

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    return (T) Proxy.newProxyInstance(
+        clazz.getClassLoader(),
+        new Class[]{clazz},
+        handler);
+  }
+
+  @Override
+  public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+    // todo: special case for store(Workflow workflow) -> storeWorkflow
```
Comment (bergman) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java]: so, rename that function or pass argument type list to instumentation?

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    final String operation = method.getName();
+
+    final Instant t0 = time.get();
+    final Object ret = method.invoke(delegate, args);
+    final long durationMillis = t0.until(time.get(), ChronoUnit.MILLIS);
+
+    if (delegate instanceof Storage) {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java]: Any thoughts subclassing `MetricsStats` so we don't need to do `instanceof` here?

Comment (rouzwawi) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java]: I think this check is lightweight enough over having to split up those methods into different classes.

Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MeteredProxy.java]: True, good enough now.

**Pull request merged**